### PR TITLE
Rename BigInteger arguments to `x` and `y` 

### LINKED ISSUE #22161: [BigInteger module] cmp and cmpabs argument name
While the whole BigInteger module might not get stabilized before 2.0 due to running out of time, there are key aspects of it we're going to prioritize getting in.  One of those key aspects is `cmp` and `cmpabs`.

1. What is the API being presented?

    `cmp` and `cmpabs` are defined like this:
Screen Shot 2023-04-25 at 9 50 16 AM

Of particular note for this discussion is the argument name `b`.  I believe this is because the argument is the second involved in the comparison, as other functions in this module use `a` and `b` as the argument names.

   - How are they intended to be used?
      - They are used for comparing the `bigint` used as the `this` argument to a second `bigint`, the `b` argument, to determine which is larger or if they are the same.  This is useful for things like sorting and priority queues.

   - How are they being used in Arkouda and CHAMPS?
      - They are not used in Arkouda
      - They are not used in CHAMPS

2. What's the history of the feature, if it already exists?
    They are based off `mpz_cmp*` and `mpz_cmpabs*` in the GMP library: https://gmplib.org/manual/Integer-Comparisons.  Those functions use `op1` and `op2` for the argument names, but since the first argument would be named `this` in our version, using `op2` for the other might look a little weird.  They've been in the current form mostly unchanged since they were first added.  The type was originally defined in the GMP module as a class in 2012, extending the GMP support to include the type which was more Chapel-like.  `cmp` itself as a name is also pretty similar to the assembly instruction of the same name.

3. What's the precedent in other languages, if they support it?
     a. Python
         - Uses `x` and `y` for the names ([cmp link](https://gmpy2.readthedocs.io/en/latest/mpfr.html#gmpy2.cmp), [cmpabs link](https://gmpy2.readthedocs.io/en/latest/overview.html#gmpy2.cmp_abs))
     b. C/C++ (see GMP library above)
     c. Rust
         - Uses `self` (`this` equivalent) and `other` ([link to Rust doc](https://doc.rust-lang.org/nightly/core/cmp/trait.Ord.html#tymethod.cmp))
     d. Swift
         - The SwiftMP package doesn't seem to provide `cmp`, only the operator versions.
     e. Julia
         - Julia seems to use `a` and `b` ([link to Julia source code](https://github.com/JuliaLang/julia/blob/master/base/gmp.jl#L238))
     f. Go
        - Go seems to use `x` and `y` ([link to Go code](https://go.dev/misc/cgo/gmp/gmp.go))

4. Are there known Github issues with the feature? Issues posted by users are most relevant, but issues encountered by developers trying to use the feature are also valuable.
    - Didn't see any

5. Are there features it is related to? What impact would changing or adding this feature have on those other features? Others features can live in the same module, but they can also live in other modules (e.g. a find method on arrays should keep the find method on strings and bytes in mind; the interface for maps should keep in mind other collection types likes lists, sets, heaps and arrays, etc.)
    - They're used to implement the comparator operators on the bigint type (e.g. `<`, `>=`, etc).  The argument names are similar to a lot of other functions in the bigint module (see the API section above).

6. How do you propose to solve the problem? Try to have multiple ideas and a recommendation of what we should do.
  
### Option 1: Rename it to `other` (this is my preferred option)
    - Pros:
      - Matches Rust
      - Clearly indicates its role in the call
      - Doesn't rely on a (missing) argument name to make sense
      - We already use `other` as an argument name for 112 functions.
        - Around half of those are in `init` or `init=`, but the other half are named functions.  Not all of these functions are 2.0 stabilized
        - These uses are spread across 37 modules, showing broad leaning toward this name when appropriate.
        - I don't have data on when we use something else instead (and `a`/`b` are pretty annoying to search for)
    - Cons:
      - Doesn't currently match the rest of the BigInteger module (but we could change that)
      - Would involve updating the function

### Option 2: Leave it as is
    - Pros:
      - Less work
      - Matches Julia
      - Consistent with rest of BigInteger module
    - Cons:
      - Not descriptive of its role in the function call
      - No visible "a"
      - Could be mistaken as using a name to reflect the type "bigint", even though all overloads with various types use the same argument name

### Option 3: Rename it to `y`
    - Pros:
      - Matches Python and Go
    - Cons:
      - Would involving updating the function
      - No visible "x"
      - Doesn't currently match the rest of the BigInteger module (but we could change that)

### PR DESCRIPTION
**Pull request by jabraham17**
Renames remaining BigInteger arguments to `x` and `y`, unless they imparted a special meaning of their functionality (like `n` or `numer`). This unifies argument names for differing types in overloaded functions. 

Implements changes from https://github.com/chapel-lang/chapel/issues/19005#issuecomment-1634712958

## Summary of changes
- change arguments of `init(num)` and `init(str)` to `init(x)`
  - also changed `init=`, but that should not be a user facing change 
- change arguments of `root(result, a, b)` to `root(result, x, y)`
- change arguments of `sqrt(result, a)` to `sqrt(result, x)`
- change arguments of `invert(result, a)` to `invert(result, x)`
- change arguments of `add(result, a, b)` to `add(result, x, y)`
- change arguments of `sub(result, a, b)` to `sub(result, x, y)`
- change arguments of `mul(result, a, b)` to `mul(result, x, y)`
- change arguments of `neg(result, a)` to `neg(result, x)`
- change arguments of `abs(result, a)` to `abs(result, x)`
- change arguments of `cmp(b)` to `cmp(x)`
- change arguments of `cmpabs(b)` to `cmp(x)`
- change arguments of `and(result, a, b)` to `and(result, x, y)`
- change arguments of `com(result, a)` to `com(result, x)`
- change arguments of `xor(result, a, b)` to `xor(result, x, y)`
- change arguments of `set(num)`, `set(a)`, and `set(str)` to `set(x)`
- change arguments of `swap(a)` to `swap(x)`
- updates and extends documentation for all affected functions

## New tests
- extends `test/deprecated/BigInteger/argumentNames.chpl`

## Testing
- [x] paratest with futures
- [x] paratest + gasnet
- [x] built and checked docs

[Reviewed by @bmcdonald3]

closes #22161
closes cray/chapel-private#5100

### CODE REVIEW

File: modules/standard/BigInteger.chpl [hunk 1]
```diff
@@ -4593,156 +4831,226 @@ module BigInteger {
...
-  proc bigint.cmpabs(b: uint) : int {
-    const b_ = b.safeCast(c_ulong);
+  /* See :proc:`~bigint.cmpabs` */
+  proc bigint.cmpabs(x: real) : int {
+    const x_ = x : c_double;
     var   ret: c_int;
 
     if compiledForSingleLocale() {
-      ret = mpz_cmpabs_ui(this.mpz, b_);
+      ret = mpz_cmpabs_d(this.mpz, x_);
```
Comment (bmcdonald3) [inline modules/standard/BigInteger.chpl:4994]: Is this `mpz_cmpabs_d` from `mpz_cmpabs_ui` change a case we had wrong before?

**Review by bmcdonald3 — APPROVED** [review]
Looks good to me, just a few questions; feel free to merge without another review.

File: modules/standard/BigInteger.chpl [hunk 2]
```diff
@@ -2565,6 +2663,29 @@ module BigInteger {
     }
   }
 
+  /* Set the value of ``result`` to the inverse of ``a`` modulo ``b``
+
+     .. note::
+        If an inverse does not exist, an :class:`InversionError` will be thrown,
+        and the value of ``result`` will be left undefined
```
[the remark this answers is not in the document]
Comment (jabraham17) [reply modules/standard/BigInteger.chpl:2670]: I converted the note into a `:throws:` in the chpldoc

File: modules/standard/BigInteger.chpl (hunk 1 above)
Comment (jabraham17) [reply modules/standard/BigInteger.chpl:4994]: There should be no change to this, it was correct before.

I think the way the diff is formatted is misleading, I had to double check several times.

**Pull request merged**